app.js

```diff
@@ -2,12 +2,12 @@ const express = require('express');
 const mongoose = require('mongoose');
 const app = express();
 const methodOverride = require('method-override');
-const router = require('./routes/tasks');
+const router = require('./src/routes/tasks');
 
 app.use(methodOverride('_method' , {method: ['POST' , 'GET']}));
```

Comment:
I read about this middleware , the reason is to use **UPDATE** and **DELETE** operation 
but really I didn't understand exactly why , If you have any best practices or alternative suggestions for performing these operations, I would appreciate it if you could share them with me so that I can learn more .

Reply:
@RodainaTariq  I'm simply asking to understand if you're aware of its usage, my question isn't meant to say you're wrong.
Essentially, if your client ((like a browser or mobile device) doesn't natively support sending `PUT`(not update) nor `DELETE` methods in its headers, then that's where this middleware can come in handy.
See [this answer](https://stackoverflow.com/a/23643894/15107393). 

Take HTML [forms as an example ](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fs-method).
HTML forms used to only support `GET` and `POST` methods, and when people needed to use `PUT` or `DELETE`, they found workarounds to use them as sub-types of `POST` requests. [source](https://medium.com/@carlospineda/why-no-methods-for-put-delete-in-html-f483b66d8874)

Another approach to send a `DELETE` request to the server, you can use the native [`fetch` API](https://jsonplaceholder.typicode.com/guide/). Here's an example:
```js
function handleDelete() {
  fetch('https://jsonplaceholder.typicode.com/posts/1', { method: 'DELETE' })
    .then(() => {
      // do something after the delete request is successful
    });
}
```
If you attach `handleDelete()` to a delete button, you'll be able to send a `DELETE` verb in your header's method when the button is clicked.
That being said, if using this middleware made things work for you then that's totally cool! There's no right or wrong way to do it.